components/Common/CodeBox/CodeTabs/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { ArrowUpRightIcon } from '@heroicons/react/24/solid';
+import * as TabsPrimitive from '@radix-ui/react-tabs';
+import type {
+  ComponentProps,
+  FC,
+  PropsWithChildren,
+  ReactElement,
+} from 'react';
+import { useMemo, useState } from 'react';
+
+import Tabs from '@/components/Common/Tabs';
+import { Link } from '@/navigation.mjs';
+
+import styles from './index.module.css';
+
+type CodeTabsProps = Pick<
+  ComponentProps<typeof Tabs>,
+  'tabs' | 'onValueChange' | 'defaultValue'
+> & {
+  linkUrl?: string;
+  linkText?: string;
+};
+
+export const CodeTabs: FC<PropsWithChildren<CodeTabsProps>> = ({
+  children,
+  linkUrl,
+  linkText,
+  ...props
+}) => (
+  <Tabs
+    {...props}
+    className={styles.root}
+    headerClassName={styles.header}
+    addons={
+      linkUrl &&
+      linkText && (
+        <Link className={styles.link} href={linkUrl}>
+          {linkText}
+          <ArrowUpRightIcon className={styles.icon} />
+        </Link>
+      )
+    }
+  >
+    {children}
+  </Tabs>
+);
+
+type MDXCodeTabsProps = {
+  children: Array<ReactElement>;
+  languages: string;
+  displayNames?: string;
+};
+
+export const MDXCodeTabs: FC<MDXCodeTabsProps> = props => {
```

Comment:
If MDXCodeTabs, and MDXCodeBox were inside `components/MDX` instead of `components/Common` :) as they are MDX specific.

Reply:
Also, the MDXCodeTabs component should be a client-side component by using `"use client";` the other ones don't need to.